File: JS/Benchmark/test/test-mandelbrot.js
// test-mandelbrot.js
// test for Mandelbrot set module
//
// vim: ft=javascript sw=4 ts=4
//

'use strict';

var m = require('../mandelbrot.js');
var expect = require('expect.js');

describe('Mandelbrot module', function () {

	it('should export mandelbrot function', function () {
		expect(m.mandelbrot).to.be.a('function');
	});
});

File: Ruby/Benchmark/test/test_mandelbrot.rb
# vim: ft=ruby sw=2 ts=2
#
require 'minitest/autorun'

require_relative '../lib/Benchmark/mandelbrot.rb'

include Benchmark

describe Mandelbrot, 'Calculate Mandelbrot set' do
  it 'mandelbrot should return an array of rgb pixel values' do
    mandel_set = Mandelbrot.new(640, 480, 0.0, 0.0, 0.5)
    _(mandel_set.width).must_equal 640
    _(mandel_set.height).must_equal 480
  end

  it 'should write image data to a PPM file' do
    mandel_set = Mandelbrot.new(640, 480, -0.5, 0.0, 4.0 / 640)
    file_name = './test_data/mandelbrot.ppm'
    File.delete file_name if File.exist? file_name
    mandel_set.to_ppm(file_name)
    _(File.exist?(file_name)).must_equal true
  end
end
